workflows/iuf/operations/management-nodes-rollout/management-worker-nodes-reboot.yaml

```
@@ -0,0 +1,547 @@
+          - name: global_params
+      dag:
+        tasks:
+          - name: get-reboot-set
+            templateRef:
+              name: ssh-template
+              template: shell-script
+            arguments:
+              parameters:
+              - name: dryRun
+                value: "{{$.DryRun}}"
+              - name: mediaHost
+                value: "{{=jsonpath(inputs.parameters.global_params, '$.input_params.media_host')}}"                
+              - name: scriptContent
+                value: |
+                  sets="{{inputs.parameters.sets}}"
+                  current_count="{{inputs.parameters.counter}}"
+                  IFS=' ' read -ra set_array  <<< "$sets"
+                  reboot_set=${set_array[$current_count]}
+                  if [[ -z $reboot_set ]]; then
+                    echo "NOTICE This step received no nodes to reboot. Continuing without rebooting any worker nodes"
+                    exit 0
+                  fi
+                  echo $reboot_set | jq -R 'split(",")'
+          - name: drain-reboot-set
+            dependencies:
+              - get-reboot-set
+            template: drain-set
+            arguments:
+              parameters:
+                - name: global_params
+                  value: "{{inputs.parameters.global_params}}"
+                - name: targetNcns
+                  value: "{{tasks.get-reboot-set.outputs.result}}"
+          - name: reboot-and-post-check
+            dependencies:
+              - drain-reboot-set
+            template: reboot-worker
+            arguments:
+              parameters:
+                - name: global_params
+                  value: "{{inputs.parameters.global_params}}"
+                - name: targetNcn
+                  value: "{{item}}"
+            withParam: "{{tasks.get-reboot-set.outputs.result}}"
+          - name: platform-health-checks
+            dependencies: 
+              - reboot-and-post-check
+            templateRef:
+              name: ncn-reboot-common
+              template: platform-health-checks
+            arguments:
+              parameters:
+                - name: dryRun
+                  value: "{{$.DryRun}}"
+          - name: check-bgp-session
+            dependencies: 
+              - reboot-and-post-check
+            templateRef:
+              name: ssh-template
+              template: shell-script
+            arguments:
+              parameters:
+                - name: dryRun
+                  value: "{{$.DryRun}}"
+                - name: scriptContent
+                  value: |
+                    echo "INFO Validating BGP peering sessions"
+                    result=$(GOSS_BASE=/opt/cray/tests/install/ncn goss \
+                      -g /opt/cray/tests/install/ncn/tests/goss-switch-bgp-neighbor-aruba-or-mellanox.yaml \
+                      --vars=/opt/cray/tests/install/ncn/vars/variables-ncn.yaml validate 2>&1)
+                    if [[ $? -ne 0 ]]; then
```

These can be combined, the `result` isn't used:
```suggestion
                    if ! GOSS_BASE=/opt/cray/tests/install/ncn goss \
                      -g /opt/cray/tests/install/ncn/tests/goss-switch-bgp-neighbor-aruba-or-mellanox.yaml \
                      --vars=/opt/cray/tests/install/ncn/vars/variables-ncn.yaml validate >/dev/null 2>&1; then
```